xla/service/gpu/gpu_hlo_schedule.cc

```diff
@@ -258,6 +259,7 @@ SchedulerConfig GetSchedulerConfig(int64_t memory_limit) {
   config.aggressive_scheduling_policies = true;
   config.schedule_send_recvs = true;
   config.memory_limit = memory_limit;
+  config.global_collective_scheduling_resource = collective_resource;
```

Comment:
How will this work with other configurations options above like `all_reduce_overlap_limit`?

Reply:
the other overlap limits are not used anywhere so im not sure how this will interact with the other ones. They are probably used in other backends. IIRC the other limits controls individual collective scheduling resource but not the mix-and-match resource limit.

Reply:
Thanks! Would it make sense to CHECK fail if there are different (>1) values provided for these fields AND global_collective_scheduling_resource > 1?

Reply:
I think having a check makes sense, though we should still allow these individual values to be set > 1 together with global resource, but each individual value cannot exceed the global resource.